src/Components/SlateEditor/SlateEditor.js

```diff
@@ -124,6 +166,70 @@ export default function SlateEditor() {
     },
   };
 
+  const getCoordinates = () => {
+    let x, y;
+    const selection = window.getSelection();
+    if (selection.rangeCount !== 0) {
+      const range = selection.getRangeAt(0).cloneRange();
+      range.collapse(false);
+      const rect = range.getClientRects()[0];
+      if (rect) {
+        x = rect.top;
+        y = rect.left;
+      }
+    }
+    return { top: x > 120 ? x - 150 : x + 20, left: y };
+  };
+
+  const closeMenu = useCallback(() => {
+    setShowMenu(false);
+    document.removeEventListener("click", closeMenu);
+  }, []);
+
+  const openMenu = useCallback(() => {
+    const position = getCoordinates();
+    setCoordinates(position);
+    setShowMenu(true);
+    document.addEventListener("click", closeMenu);
+  }, [closeMenu]);
+
+  const renderMenu = () => {
+    return (
+      <Menu
+        className="editor__menu"
+        style={coordinates}
+        ref={menuFocus}
+        onKeyDown={(event) => {
+          if (event.key === " " || event.key === "Backspace") {
+            closeMenu();
+            ReactEditor.focus(editor);
+            event.preventDefault();
+          }
+
+          if (event.key === "Enter") {
+            event.preventDefault();
+          }
+        }}
+      >
+        {menuOptions.map((item, key) => {
```

Comment:
make this a separate block and use that block inside here - makes the code more cleaner